docs/source/topics/scheduler_template.py

```diff
@@ -0,0 +1,182 @@
+# -*- coding: utf-8 -*-
+"""Template for a scheduler plugins."""
+import datetime
+import logging
+
+from aiida.common.escaping import escape_for_bash
+from aiida.schedulers import Scheduler, SchedulerError, SchedulerParsingError
+from aiida.schedulers.datastructures import (JobInfo, JobState, MachineInfo, NodeNumberJobResource, ParEnvJobResource)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+_MAP_SCHEDULER_AIIDA_STATUS = {
+    'SET_ME_STATUS_1': JobState.RUNNING,
+    'SET_ME_STATUS_5': JobState.QUEUED,
+    'SET_ME_STATUS_2': JobState.DONE,
+    'SET_ME_STATUS_3': JobState.QUEUED_HELD,
+    'SET_ME_STATUS_4': JobState.UNDETERMINED,
+    'SET_ME_STATUS_6': JobState.SUSPENDED,
+}
+
+
+class TemplateJobResource(NodeNumberJobResource):  # or TemplateJobResource(ParEnvJobResource)
+    """Template class for job resources."""
+
+    @classmethod
+    def validate_resources(cls, **kwargs):
+        """Validate the resources against the job resource class of this scheduler.
+
+        This extends the base class validator.
+
+        :param kwargs: dictionary of values to define the job resources
+        :return: attribute dictionary with the parsed parameters populated
+        :raises ValueError: if the resources are invalid or incomplete
+        """
+        resources = super().validate_resources(**kwargs)
+
+        # Put some validation code here.
+
+        return resources
```

Comment:
```suggestion
    def validate_resources(cls, **kwargs):
        """Validate the resources against the job resource class of this scheduler.

        :param kwargs: dictionary of values to define the job resources
        :raises ValueError: if the resources are invalid or incomplete
        :return: optional tuple of parsed resource settings
        """

    def accepts_default_mpiprocs_per_machine(cls):
        """Return True if this subclass accepts a `default_mpiprocs_per_machine` key, False otherwise."""

    def get_tot_num_mpiprocs(self):
        """Return the total number of cpus of this job resource."""
```